actionpack/test/controller/parameters/accessors_test.rb

```diff
@@ -419,4 +418,28 @@ class ParametersAccessorsTest < ActiveSupport::TestCase
     @params.dig(:person, :addresses)[0] = { city: "Boston", state: "Massachusetts" }
     assert_equal "Boston", @params.dig(:person, :addresses, 0, :city)
   end
+
+  test "has_value? works with hash argument after conversion" do
+    without_deprecated_parameters_hash_equality do
+      params = ActionController::Parameters.new(foo: { bar: "baz" })
+      assert params.has_value?("bar" => "baz")
+      params[:foo] # converts value to AC::Params
+      assert params.has_value?("bar" => "baz")
+    end
+  end
+
+  test "has_value? works with AC:Parameters argument before conversion" do
+    without_deprecated_parameters_hash_equality do
```

Comment:
I don't agree that this should raise a deprecation warning. The developer is doing nothing wrong in this usage and it only raises because the value isn't converted yet. We should change it so this doesn't raise a deprecation warning.